application-load-balancer/main.tf

```diff
@@ -166,3 +170,260 @@ output "cert-arn" {
 output "alb-arn" {
   value = aws_lb.this.arn
 }
+
+
+## This section configures WAF on ALB to attach security token.
+
+data "aws_caller_identity" "current" {}
+
+# Random password for the secret value
+resource "random_password" "origin-secret" {
+  length           = 32
+  special          = false
+  override_special = "_%@"
+}
+
+resource "aws_wafv2_web_acl" "waf-acl" {
+  # checkov:skip=CKV2_AWS_31: Ensure WAF2 has a Logging Configuration to be done new ticket
+  # checkov:skip=CKV_AWS_192: AWSManagedRulesKnownBadInputsRuleSet handles on the CDN
+  depends_on = [random_password.origin-secret]
+
+  name        = "${var.application}-${var.environment}-ACL"
+  description = "CloudFront Origin Verify"
+  scope       = "REGIONAL"
+
+  default_action {
+    block {} # Action to perform if none of the rules contained in the WebACL match
+  }
+
+  visibility_config {
+    cloudwatch_metrics_enabled = true
+    metric_name                = "${var.application}-${var.environment}-XOriginVerify"
+    sampled_requests_enabled   = true
+  }
+
+  rule {
+    name     = "${var.application}-${var.environment}-XOriginVerify"
+    priority = "0"
+
+    action {
+      allow {}
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "${var.application}-${var.environment}-XMetric"
+      sampled_requests_enabled   = true
+    }
+
+    statement {
+      byte_match_statement {
+        field_to_match {
+          single_header {
+            name = "x-origin-verify"
+          }
+        }
+        positional_constraint = "EXACTLY"
+        search_string         = random_password.origin-secret.result
```

Comment:
Just checking I understand. This is just setting a random value initially, which will be overwritten as soon as the lambda runs?

Reply:
yes, it should get overwritten when we rotate, which should be as asoon as the secret rotation resource is provisioned.